### PR TITLE
[Lint] OSM: const'ify and smartptr'ify

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

File: port/cpl_vsi_mem.cpp [hunk 1]
```diff
@@ -266,13 +265,21 @@ bool VSIMemFile::SetLength(vsi_l_offset nNewLength)
             return false;
         }
 
-        const vsi_l_offset nNewAlloc = (nNewLength + nNewLength / 10) + 5000;
+        // If the first allocation is 1 MB or above, just take that value
+        // as the one to allocate
+        // Otherwise slightly reserve more to avoid too frequent reallocations.
+        const vsi_l_offset nNewAlloc =
```
Comment (elpaso) [inline port/cpl_vsi_mem.cpp:271]: How are these changes related with OSM changes?

**Review by elpaso — APPROVED** [review]

File: port/cpl_vsi_mem.cpp (hunk 1 above)
Comment (rouault) [reply port/cpl_vsi_mem.cpp:271]: yes slight "optimization" I discovered when realizing that the OSM code didn't actually pre-allocate the /vsimem/ file as it was supposed to do
